beater/beater.go

```diff
@@ -431,6 +494,34 @@ func (s *serverRunner) run() error {
 	return publisher.Stop(s.backgroundContext)
 }
 
+type dynamicConfig struct {
```

Comment:
I think I led you astray: this mirrors beater/config, so probably not very maintainable. How about instead we just zero out the dynamic config, like you're doing already, to check if a restart is needed. If so restart, and otherwise call updateDynamicConfig with the `*beater/config.Config`?

Reply:
I think this is a solid improvement :ok_hand: